webapp/src/main/webapp/larva/assets/lib.js

```diff
@@ -396,9 +402,17 @@ function indentCompare(sources, result)
 {
 	var s = eval(sources);
 	
-	for (var i = 0; i < s.length; i++)
-		xmlFormat(s[i]);
-		
+	for (var i = 0; i < s.length; i++){
+		var elementId = s[i];
+		var text = $(elementId).value;
+		// if text is xml
+		if(text.startsWith("<") || text.startsWith(escapeChars("<"))){
+			xmlFormat(elementId);
+		} else if ((text.startsWith("{") && text.endsWith("}")) || (text.startsWith("[") && text.endsWith("]"))) { // TODO: something smarter for detecting json like text
+			jsonFormat(elementId);
+		}
```

Comment:
When the formatting fails, e.g. due to a syntax error, something like the original text should be displayed.